projects/web3/eth_wallet/ta/src/main.rs

```diff
@@ -83,21 +101,27 @@ fn create_wallet(_input: &proto::CreateWalletInput) -> Result<proto::CreateWalle
     })
 }
 
-fn remove_wallet(input: &proto::RemoveWalletInput) -> Result<proto::RemoveWalletOutput> {
+fn remove_wallet(
+    db_client: &SecureStorageClient,
+    input: &proto::RemoveWalletInput,
+) -> Result<proto::RemoveWalletOutput> {
     dbg_println!("[+] Removing wallet: {:?}", input.wallet_id);
 
-    delete_from_secure_storage(input.wallet_id.as_bytes())?;
+    db_client.delete_entry::<Wallet>(&input.wallet_id.to_string())?;
     dbg_println!("[+] Wallet removed");
 
     Ok(proto::RemoveWalletOutput {})
 }
 
-fn derive_address(input: &proto::DeriveAddressInput) -> Result<proto::DeriveAddressOutput> {
-    let secure_object = load_from_secure_storage(input.wallet_id.as_bytes())
+fn derive_address(
+    db_client: &SecureStorageClient,
+    input: &proto::DeriveAddressInput,
+) -> Result<proto::DeriveAddressOutput> {
+    let wallet = db_client
+        .get::<Wallet>(&input.wallet_id.to_string())
```

Comment:
&input.wallet_id

Reply:
```
    pub fn get<V>(&self, key: &V::Key) -> Result<V>
    where
        V: Storable + serde::de::DeserializeOwned,
        V::Key: Into<String> + Clone,
```
The StorageKey type (`V::Key`) should impl `Into<String>`, since the Uuid is from the third-party crate `uuid`, we cannot impl trait for it.
Here I convert the Uuid into String and use the String as StorageKey type. But we need to `&input.wallet_id.to_string()`. To eliminate this maybe we can define a wrapper struct `UuidKey(Uuid)` which is also not elegant, or our abstraction needs to be improved for this case.

Reply:
Fixed in the latest commit: use ToString instead of Into<String>.